# HG changeset patch
# User Nils Bruin <nbruin@sfu.ca>
# Date 1299014848 28800
# Node ID 56afff4b15b4660e2332d9b06be2bf747d82f7dc
# Parent  13499b4b5d702c424194898f739c2a230f5a641e
make max_to_sr return RealDoubleElement rather than float

diff -r 13499b4b5d70 -r 56afff4b15b4 sage/interfaces/maxima_lib.py
--- sage/interfaces/maxima_lib.py	Tue Mar 01 09:41:23 2011 -0800
+++ sage/interfaces/maxima_lib.py	Tue Mar 01 13:27:28 2011 -0800
@@ -725,6 +725,8 @@
             return maxima(expr).ecl()
     
 # This goes from EclObject to SR
+import sage.rings.real_double
+
 def max_to_sr(expr):
     if expr.consp():
         op_max=caar(expr)
@@ -746,5 +748,8 @@
             max_sym_dict[expr]=sage_symbol
         return max_sym_dict[expr]
     else:
-        return expr.python()
+        e=expr.python()
+        if isinstance(e,float):
+            return sage.rings.real_double.RealDoubleElement(e)
+        return e
 
